### ISSUE TITLE
Switching off the page and test history

### ISSUE THREAD

**Issue by afmg-ekoeppen** [unclebob/fitnesse#1488]
The creation of the test and page history is not relevant for my use case. It would be nice to disable this completely in the configuration so that no zip folders are created in the fitnesse folder structure. 

A possible solution strategy that I could well imagine would be. Define variables in the root page.

```
!define test.history.days {0}
!define page.history.days {0}
````

Or something similar.

**Comment by six42** [comment]
Are you aware of the existing nohistory parameter?
https://fitnesse.org/FitNesse/UserGuide/AdministeringFitNesse/RestfulServices.html

**Comment by afmg-ekoeppen** [comment]
> Are you aware of the existing nohistory parameter? https://fitnesse.org/FitNesse/UserGuide/AdministeringFitNesse/RestfulServices.html

Thank you. Yes, I am aware of that parameter, but it doesn't help me for my application.

**Comment by six42** [comment]
Hi Emil
Can you explain more why doesn't it helps?
What are your requirements?
How do you run your tests?
Do you want to have it switched off for the whole Wiki or just some pages, with or without child pages?

**Comment by afmg-ekoeppen** [comment]
> Hi Emil Can you explain more why doesn't it helps? What are your requirements? How do you run your tests? Do you want to have it switched off for the whole Wiki or just some pages, with or without child pages?

In my case, the history is mapped with a version management system. I work with a checkout on my local PC and host the FitNesse Wiki here. I run the tests on my PC in the browser. I edit the tests within an IDE and work in the FitNesse file system. For the wiki on my local system, I would like to be able to switch off the entire history for all tests in order to clean the file system of the generated zip folders. At the moment, I have to clean my checkout manually.

**Comment by six42** [comment]
Did you tried to set in plugins. properties:

test.history.days=0

This seems to work for me.

**Comment by fhoeben** [comment]
@afmg-ekoeppen I also never want to use these files (with my IDE or otherwise), but they have never really bothered me since I have the following in my project's `.gitignore`:
```
wiki/FitNesseRoot/**/*.zip
wiki/FitNesseRoot/files/testResults
wiki/FitNesseRoot/files/testProgress
wiki/FitNesseRoot/ErrorLogs
wiki/FitNesseRoot/FitNesse
wiki/FitNesseRoot/RecentChanges*
wiki/FitNesseRoot/ReRunLastFailures*.wiki
updateDoNotCopyOverList
updateList
properties
```

I never look in the `files` directory with my IDE. Is there a particular reason why you really don't want them there, or is it more a general desire for cleanliness?

**Comment by afmg-ekoeppen** [comment]
> @afmg-ekoeppen I also never want to use these files (with my IDE or otherwise), but they have never really bothered me since I have the following in my project's `.gitignore`:
> 
> ```
> wiki/FitNesseRoot/**/*.zip
> wiki/FitNesseRoot/files/testResults
> wiki/FitNesseRoot/files/testProgress
> wiki/FitNesseRoot/ErrorLogs
> wiki/FitNesseRoot/FitNesse
> wiki/FitNesseRoot/RecentChanges*
> wiki/FitNesseRoot/ReRunLastFailures*.wiki
> updateDoNotCopyOverList
> updateList
> properties
> ```
> 
> I never look in the `files` directory with my IDE. Is there a particular reason why you really don't want them, or is it more a general desire for cleanliness?

Yes, it is my general desire for cleanliness. My ignore file looks very similar.

**Comment by afmg-ekoeppen** [comment]
> @afmg-ekoeppen I also never want to use these files (with my IDE or otherwise), but they have never really bothered me since I have the following in my project's `.gitignore`:
> 
> ```
> wiki/FitNesseRoot/**/*.zip
> wiki/FitNesseRoot/files/testResults
> wiki/FitNesseRoot/files/testProgress
> wiki/FitNesseRoot/ErrorLogs
> wiki/FitNesseRoot/FitNesse
> wiki/FitNesseRoot/RecentChanges*
> wiki/FitNesseRoot/ReRunLastFailures*.wiki
> updateDoNotCopyOverList
> updateList
> properties
> ```
> 
> I never look in the `files` directory with my IDE. Is there a particular reason why you really don't want them, or is it more a general desire for cleanliness?

Yes, it is my general desire for cleanliness.

**Comment by six42** [comment]
Hi Emil,

nothing will be written in the folder FitNesseRoot/files/testResults/ if you do the setting:  test.history.days=0
TestHistory and PageHistory both get the data from this folder.

The zip files are created each time you edit a wiki page. They are not related to testing.
To disable them do this in plugin.properties

VersionsController=fitnesse.wiki.fs.SimpleFileVersionsController

**Comment by afmg-ekoeppen** [comment]
> VersionsController=fitnesse.wiki.fs.SimpleFileVersionsController

**Many thanks for the effort! My problem is solved.** 

Defining the variable `test.history.days=0` in `plugin.properties` did nothing for me. I have now defined it in the `root page` with `!define test.history.days {0}`, that works well. The other variable I have defined the other variable as suggested and it works.

**Comment by six42** [comment]
Good that version control works for you.
Strange that test.history.days=0 did not work for you. 
Maybe one thing. You need to restart after you modified plugin.properties.

**Comment by afmg-ekoeppen** [comment]
> Good that version control works for you. Strange that test.history.days=0 did not work for you. Maybe one thing. You need to restart after you modified plugin.properties.

I write the properties before the server is launched locally.

**Comment by six42** [comment]
And you had removed/ overwritten the existing entry?
